eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/io/AssemblyNameRetriever.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * The ENA webservices can return an xml that starts like this:
+ * <pre>
+ * {@code
+ * <?xml version="1.0" encoding="UTF-8"?>
+ * <ROOT request="GCA_000001405.28&amp;display=xml">
+ * <ASSEMBLY accession="GCA_000001405.28" alias="GRCh38.p13" center_name="Genome Reference Consortium">
+ * ...
+ * }
+ * </pre>
+ *
+ * However, the alias is optional, according to
+ * https://github.com/enasequence/schema/blob/master/src/main/resources/uk/ac/ebi/ena/sra/schema/ENA.assembly.xsd#L32
+ * which defines an AssemblyType as an extension of ObjectType, and
+ * https://github.com/enasequence/schema/blob/master/src/main/resources/uk/ac/ebi/ena/sra/schema/SRA.common.xsd#L21
+ * defines alias as optional attribute of ObjectType.
+ */
+public class AssemblyNameRetriever {
+
+    private static final String ENA_ASSMEBLY_URL_FORMAT_STRING = "https://www.ebi.ac.uk/ena/data/view/%s";
+
+    private static final String ENA_ASSMEBLY_XML_DISPLAY_SUFFIX = "&display=xml";
+
+    private String assemblyAccession;
+
+    private String assemblyName;
+
+    public AssemblyNameRetriever(String assemblyAccession) {
+        this.assemblyAccession = assemblyAccession;
+        this.assemblyName = fetchAssemblyName(assemblyAccession);
+    }
+
+    private String fetchAssemblyName(String assemblyAccession) {
+        String url = buildAssemblyUrl(assemblyAccession) + ENA_ASSMEBLY_XML_DISPLAY_SUFFIX;
+
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(EnaAssemblyXml.class);
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            EnaAssemblyXml enaAssembly = (EnaAssemblyXml) unmarshaller.unmarshal(new URL(url));
+            if (enaAssembly.getAssembly() == null) {
+                return null;
+            } else {
+                return enaAssembly.getAssembly().getAlias();
+            }
+        } catch (JAXBException | MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String buildAssemblyUrl(String assemblyAccession) {
+        return String.format(ENA_ASSMEBLY_URL_FORMAT_STRING, assemblyAccession);
+    }
+
+    public String getAssemblyAccession() {
+        return assemblyAccession;
+    }
+
+    public Optional<String> getAssemblyName() {
+        if (assemblyName == null) {
+            return Optional.empty();
+        } else {
+            return Optional.of(assemblyName);
+        }
+    }
+
+    public String buildAssemblyUrl() {
+        return buildAssemblyUrl(assemblyAccession);
+    }
+
+    @XmlRootElement(name = "ROOT")
+    static class EnaAssemblyXml {
+
+        @XmlElement(name = "ASSEMBLY")
+        private Assembly assembly;
+
+        public EnaAssemblyXml() {
+        }
+
+        public Assembly getAssembly() {
+            return assembly;
+        }
+
+        static class Assembly {
+
+            @XmlAttribute
+            private String alias;
```

Comment:
There is a `NAME` attribute in the XML response. Is it better to use the alias instead of the name?
```
<ASSEMBLY accession="GCA_000004515.3" alias="Glycine_max_v2.0" center_name="US DOE Joint Genome Institute (JGI-PGF)">
[...]
    <NAME>Glycine_max_v2.0</NAME>
```

Reply:
Ensure that NAME attribute is mandatory as part of the XSD and you can also check if it has expected values for a few species.

Reply:
I checked that it's not optional and it gives the same results as the alias for all our species, so I'll use the NAME